tools/changelogger/src/render.rs

```diff
@@ -0,0 +1,576 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::entry::{ChangeSet, ChangelogEntries, ChangelogEntry};
+use anyhow::{Context, Result};
+use clap::Parser;
+use ordinal::Ordinal;
+use serde::Serialize;
+use smithy_rs_tool_common::changelog::{
+    Changelog, HandAuthoredEntry, Reference, SdkModelChangeKind, SdkModelEntry,
+};
+use smithy_rs_tool_common::git::{find_git_repository_root, Git, GitCLI};
+use std::env;
+use std::fmt::Write;
+use std::fs;
+use std::path::PathBuf;
+use time::OffsetDateTime;
+
+pub const EXAMPLE_ENTRY: &str = r#"
+# Example changelog entries
+# [[aws-sdk-rust]]
+# message = "Fix typos in module documentation for generated crates"
+# references = ["smithy-rs#920"]
+# meta = { "breaking" = false, "tada" = false, "bug" = false }
+# author = "rcoh"
+#
+# [[smithy-rs]]
+# message = "Fix typos in module documentation for generated crates"
+# references = ["smithy-rs#920"]
+# meta = { "breaking" = false, "tada" = false, "bug" = false }
+# author = "rcoh"
+"#;
+
+pub const USE_UPDATE_CHANGELOGS: &str =
+    "<!-- Do not manually edit this file. Use the `changelogger` tool. -->";
+
+fn maintainers() -> Vec<&'static str> {
+    include_str!("../smithy-rs-maintainers.txt")
+        .lines()
+        .collect()
+}
+
+#[derive(Parser, Debug, Eq, PartialEq)]
+pub struct RenderArgs {
+    /// Which set of changes to render
+    #[clap(long, action)]
+    pub change_set: ChangeSet,
+    /// Whether or not independent crate versions are being used (defaults to false)
+    #[clap(long, action)]
+    pub independent_versioning: bool,
+    /// Source changelog entries to render
+    #[clap(long, action)]
+    pub source: Vec<PathBuf>,
```

Comment:
Can we make this a required parameter please - thanks

Reply:
I think you mean have a minimum of one? Good call out. I'll take a look at what clap offers for this.

Reply:
Looks like `required(true)` does the job.